library(shiny)

ui <- fluidPage(
  
  # App title ----
  titlePanel("One Stop-shop to Explore Single Cell Reources/Tools"),
  
  # Sidebar layout with input and output definitions ----
    mainPanel(
      tabsetPanel(type = "tabs",
                  tabPanel("About", fluidRow(
                    p("This resource is developed with the intention to provide a one stop-shop experience to users in terms of serching single cell datasets from published studies", style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p("Please contribute to this resource via Github Issues section", a("Access GitHub", href="https://github.com/praneet1988/Explore-Single-cell-Resources-and-Databases", target="_blank"), style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p("Developed by Praneet Chaturvedi. To view other tools and contributions, please access", a("GitHub", href="https://github.com/praneet1988", target="_blank"), a("Twitter", href="https://twitter.com/praneet006", target="_blank"), style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"))),

                  tabPanel("Single Cell Resources", fluidRow(
                    p(strong("Single cell resource at HCL:"), "Single-cell analysis is a valuable tool to dissect cellular heterogeneity in complex systems. Yet, a comprehensive single-cell atlas has not been achieved for humans. Human cell landscape (HCL) used single-cell mRNA sequencing to determine the cell-type composition of all major human organs and constructed a scheme for the human cell landscape (HCL):", a('Explore', href="http://bis.zju.edu.cn/HCL/index.html", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Single cell resource at UCSC:"), "Single cell experiments measure tens of thousands of gene loci in thousands of cells at the same time. That resulting data is challenging to visualize. The UCSC Genome Browser is adding published datasets as tracks and an entirely new data exploration tool:", a('Explore', href="https://genome.ucsc.edu/singlecell.html", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Single Cell Portal at Broad:"), "Single cell portal currently features data for 218 studies and ~6 million cells:", a('Explore', href="https://singlecell.broadinstitute.org/single_cell", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("PanglaoDB:"), "PanglaoDB is a database for the scientific community interested in exploration of single cell RNA sequencing experiments from mouse and human. We collect and integrate data from multiple studies and present them through a unified framework:", a('Explore', href="https://panglaodb.se/", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Single Cell Expression Atlas (EBI):"), "Single Cell Expression Atlas hosted at European Bioinformatics institute features data across 14 species, 151 studies and ~2 million cells:", a('Explore', href="https://www.ebi.ac.uk/gxa/sc/home", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Human Cell Landscape:"), "We used single-cell RNA sequencing to determine the cell type composition of major human organs and construct a basic scheme for the human cell landscape (HCL). We analysed >700,000 single cells from >50 human tissues (2-4 replicates per tissue in general) and cultures. In a global view, the complete human tissue dataset are grouped into 102 major clusters. We created this page to achieve better visualization for global view. You can view each group, each tissue and each gene by choosing the optional box on the left. Landscape view provides global view on single cell level, Marker list provides marker genes for each cluster. Gene search (Pseudocell) provides global view on pseudo-cell20 level", a('Explore', href="http://bis.zju.edu.cn/HCL/", target="_blank"), style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("SCDevDB:"), "A database for insights into single-cell gene expression profiles during human developmental processes:", a('Explore', href="https://scdevdb.deepomics.org/", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Brain Map Portal:"), "The mammalian brain is composed of many cell populations that differ based on their molecular, morphological, electrophysiological and functional characteristics. Classifying these cells into types is one of the essential approaches to defining the diversity of the brain's building blocks. This project seeks to characterize cortical diversity at the cellular level for several neuroanatomical areas in both mouse and human. Additional data will be released regularly, building toward a complete picture of cellular diversity of the brain and how this diversity is conserved across species:", a('Explore', href="https://portal.brain-map.org/atlases-and-data/rnaseq", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("SCPortalen:"), "Single-cell omics recently emerged as powerful tools to investigate heterogeneity of large populations of cells. Among others, improvements in sequencing, microscopy and microfluidic technologies let to a rapid increase of complex datasets with single-cell resolution. However, due to a lack of available platforms to easily share and integrate complex single-cell datasets, the accessibility of such datasets can be a barrier to efficient usage. The number of publication in the area of single-cell increased in the recent years. Still we are lacking public single-cell database to facilitate and enable researchers to access and explore published single-cell datasets. To address the above issue, we introduced a single-cell data integration platform. The motivation for the development of this database was to enable easy access, integration and collaboration on single-cell datasets generated by the research community worldwide. The current version of the database integrate of single-cell metadata, cell images and sequence information. As an added-value to the raw data, we performed manual curation and annotation of each dataset and conducted single-cell analysis of single-cell RNA-seq. The easy-to-use user interface of the database will enable biologist and computing researchers to search, explore, and download single-cell data and images:", a('Explore', href="http://single-cell.clst.riken.jp/", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("scRNASeqDB:"), "Single-cell RNA-Seq (scRNA-seq) are an emerging method which facilitates to explore the comprehensive transcriptome in a single cell. To provide a useful and unique reference resource for biology and medicine, we developed the scRNASeqDB database, which contains 36 human single cell gene expression data sets collected from Gene Expression Omnibus (GEO), involving 8910 cells from 174 cell groups. We also provides detailed information for gene expression of cells in different status, as well as some features, including heatmap and boxplot of gene expression, gene correlation matrix, GO and pathway annotations:", a('Explore', href="https://bioinfo.uth.edu/scrnaseqdb/", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("BioTuring:"), "Explore ~6 million cells from published works:", a('Explore', href="https://bioturing.com/", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Vascular Single Cells:"), "Explore Gene expression across vascular single cell datasets:", a('Explore', href="http://betsholtzlab.org/VascularSingleCells/database.html", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Single Cell Studies Blog by Valentine Svensson:"), "The 'Single cell studies database' lists papers in single cell transcriptomics with several fields of manually curated metadata:", a('Explore', href="http://www.nxn.se/single-cell-studies/", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"))),
                  
                  tabPanel("Single Cell Tools", fluidRow(
                    p(strong("Single Cell Tools repository maintained by Sean Davis:"), "Community-curated list of software packages and data resources for single-cell, including RNA-seq, ATAC-seq, etc.:", a('Explore', href="https://github.com/seandavi/awesome-single-cell", target="_blank"),style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"),
                    p(strong("Single Cell Tools repository maintained by Alicia Oshlack's Lab"), "The scRNA-tools database records details of software tools designed for analysing scRNA-seq data. Each tool is categorised according to the analysis tasks it can be used for. Currently we track for 627 tools over 30 categories", a('Explore', href="https://www.scrna-tools.org/", target="_blank"), style="text-align:justify;color:black;background-color:white;padding:20px;border-radius:10px;font-size:15px"))))
   )
)
server <- function(input, output) {
}
shinyApp(ui = ui, server=server)
